tsdb/series_file.go

```diff
@@ -413,7 +413,10 @@ func ParseSeriesKeyInto(data []byte, dstTags models.Tags) ([]byte, models.Tags)
 // different length and capacity to those provided.
 func parseSeriesKey(data []byte, dst models.Tags) ([]byte, models.Tags) {
 	var name []byte
-	_, data = ReadSeriesKeyLen(data)
+	sz, data := ReadSeriesKeyLen(data)
+	if sz == 0 && len(data) == 0 {
```

Comment:
I think this test should simply be `if len(data) == 0 {`

Reply:
I think you're right, I believe I've been a bit heavy-handed in also enforcing the `sz && len(data)`, as if `len(data) == 0`, then the returned size was also 0 anyway.